### PR TITLE
Remove unused class params

### LINKED ISSUE #311: Most class variables are unnecessary
A handful of the class-level variables in the project are actual class-level configuration, but most of them are shadowed by the `__init__` methods of their classes. This is untidy and could lead to confusion. I'd be happy to clean them up, unless there are objections.

### PR DESCRIPTION
**Pull request by TaylorMutch**
This PR works to resolve #311. No functionality changes are implemented, simply a cleanup of class variables that are overwritten when defined in the `__init__` methods of most classes.

### CODE REVIEW

**Comment by TaylorMutch** [comment]
@jackiekazil I'm guessing the decrease in coverage was due to the removal of code but no new tests? ='(

**Comment by rht** [comment]
This cleanup is necessary! I think the coverage issue is just due to this branch being merged with the upstream branch instead of rebased, @TaylorMutch.

File: mesa/visualization/modules/ChartVisualization.py [hunk 1]
```diff
@@ -61,6 +57,11 @@ def __init__(self, series, canvas_height=200, canvas_width=500,
             data_collector_name: Name of the DataCollector to use.
         """
 
+        if not isinstance(series, list):
+            raise TypeError("series is not a list.")
+        elif not len(series):
+            raise TypeError("series contains to elements")
```
Comment (rht) [inline mesa/visualization/modules/ChartVisualization.py]: s/to/no/

If this is fixed, this PR LGTM.

Comment (TaylorMutch) [inline mesa/visualization/modules/ChartVisualization.py]: @rht Sorry, s/to/no? I'm afraid I can't parse that...

**Comment by TaylorMutch** [comment]
@rht I undid the (unnecessary) changes I made. The PR now only removes the unnecessary class variables.

File: mesa/visualization/modules/ChartVisualization.py (hunk 1 above)
Comment (a4d442663580) [inline mesa/visualization/modules/ChartVisualization.py]: (@TaylorMutch it is a sed syntax for string substitution)

**Comment by rht** [comment]
I see...

The remaining step to make it cleaner would be to fixup all the commits into one from `git rebase -i HEAD~8`, then `git commit --amend` that one commit into "Remove unused class params".

Another note: it would be cleaner to `git pull --rebase upstream master` instead of merge. Git DAG is immutable, but it would pollute the upstream tree (harder to do git archaelogy) if edit/undo commits are included.

(If what I said is too terse, these documentation would do (two, in case if one of them is confusing): https://help.github.com/articles/about-git-rebase/, https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request.)

**Comment by TaylorMutch** [comment]
@rht Thanks for the heads up! My git-fu is a bit weak I'm afraid. I'll give it a go in the next day or two (I suspect I have performed this heresy across many projects...)

**Comment by jackiekazil** [comment]
@TaylorMutch -- let us know if you need any help with it.... we can take it offline and help walk you through it. My email is my git user name @ google's mail.

**Comment by TaylorMutch** [comment]
@rht @jackiekazil I think I did that okay... though everything looks a bit bollocked up... is this correct now?

**Comment by rht** [comment]
@TaylorMutch I don't know how to tell in a way that you can fix this in a concise number of steps. But once again, you should have used `git pull --rebase` when you used github merge, which resulted in the 20+ extra commits.

1. `git rebase -i HEAD~30`, in the editor, drop all the commits except for 
https://github.com/projectmesa/mesa/pull/381/commits/f2b737e681df6f82fef6378bcdf31dceeee8b6a7. This step is to reset the branch. Make sure to do this step correctly.
2. [optional] If you don't have the upstream remote url in `.git/config`, you have to add with
```
[remote "upstream"]
	url = git://github.com/projectmesa/mesa.git
	fetch = +refs/heads/*:refs/remotes/origin/*
```
3. `git pull --rebase upstream master` (instead of merge)
This step updates the local branch `remove-unused-class-params` to the latest master and apply `f2b737e681df6f82fef6378bcdf31dceeee8b6a7` on it.
4. `git push -f` overwrites the `origin:remove-unused-class-params`

Should you need to periodically rebase the branch, you have to do the `git pull --rebase upstream master` then `git push -f`.

**Comment by TaylorMutch** [comment]
Well, I messed that up too... So that didn't work :/

**Pull request closed without merging**